### PR TITLE
VACMS-16925: DUW Add warning box for v2

### PR DESCRIPTION
**Pull request by chriskim2311**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

## Summary
This PR adds the warning box for v2 of DUW. This is shown when they weren't sure about where they previously applied for a discharge upgrade before.

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/16935

## Testing done
Tested locally at '/discharge-upgrade-instructions/introduction'

<details><summary>Flow 1</summary> 
How_To_Apply_For_A_Discharge_Upgrade___Veterans_Affairs
</details>

<details><summary>Flow 2</summary> 
How_To_Apply_For_A_Discharge_Upgrade___Veterans_Affairs
</details>

<details><summary>Flow 3</summary> 
How_To_Apply_For_A_Discharge_Upgrade___Veterans_Affairs
</details>

<details><summary>Flow 4</summary> 
How_To_Apply_For_A_Discharge_Upgrade___Veterans_Affairs
</details>

## Screenshots

<details><summary>Flow 1</summary> 
How_To_Apply_For_A_Discharge_Upgrade___Veterans_Affairs
</details>

<details><summary>Flow 2</summary> 
How_To_Apply_For_A_Discharge_Upgrade___Veterans_Affairs
</details>

<details><summary>Flow 3</summary> 
How_To_Apply_For_A_Discharge_Upgrade___Veterans_Affairs
</details>

<details><summary>Flow 4</summary> 
How_To_Apply_For_A_Discharge_Upgrade___Veterans_Affairs
</details>

## What areas of the site does it impact?
DUW

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### CODE REVIEW

File: src/applications/discharge-wizard/helpers/index.jsx [hunk 1]
```diff
@@ -263,6 +263,12 @@ export const deriveIsAirForceAFRBAPortal = formValues =>
 
 // v2 Helpers
 
+export const determineBranchOfService = key =>
```
Comment (randimays) [inline src/applications/discharge-wizard/helpers/index.jsx:266]: Why would the Marines key map to Navy?

**Review by randimays — CHANGES_REQUESTED** [review]
Some observations:

- The below version of the warning box has a `[branch of service]` placeholder in it
Screenshot 2024-08-01 at 9 46 06 AM

- When I go through this flow:
  - Introduction
  - Navy
  - Discharged in 2023
  - DD215 that shows my discharge upgrade
  - I'm not sure what kind
  - Review page
  - Results page
  - Back up to review page
  - Edit -> Navy
  - Back to review page
  - Hit "back" button

I land back on the Navy / military branch question. At this point, the user should not be in "edit" mode, and should go back through each question, right? @thejordanwood

In general, I'm noticing that if you use the Back button at all after crossing the review page, questions are skipped when going forward (and not in "edit" mode).

- This version of the warning box has some alignment issues. This is likely coming from the web component; has a defect ticket been filed with them already?
Screenshot 2024-08-01 at 9 52 24 AM

- I saw this on one of the review screens. Wasn't this updated to Form 293 in one of the previous PRs? Please make sure that whatever you're updating in v1 of the flow is updated here as well.
Screenshot 2024-08-01 at 9 52 54 AM

- I didn't see anything in your PR about how to get to this content; is it not used?
Screenshot 2024-08-01 at 9 54 07 AM

- Looking through the `helpers` file, I think we'd benefit greatly from some documentation on the functions. Some of them may be self-explanatory, but others could benefit from a little context or perhaps a more explicit function name (such as `answerReviewLabel` - I'm not sure what this function does exactly). I mentioned in an earlier PR that we're not getting much readability in the flow by using constants such as `PREV_APPLICATION_TYPE_4` versus something like `DOC_REVIEW`. We should be mindful that this code refactor is intended to improve not only the Veteran experience using the application but also the engineer experience maintaining it.

- Please make sure to add the design Mural or any other supporting documentation to every PR you create.

File: src/applications/discharge-wizard/helpers/index.jsx (hunk 1 above)
Comment (chriskim2311) [reply src/applications/discharge-wizard/helpers/index.jsx:266]: I had to google this but apparently they do not have a discharge review board and use the Navy one.

Comment (randimays) [reply src/applications/discharge-wizard/helpers/index.jsx:266]: Oh ok - this would be helpful context for some function documentation :)

**Comment by chriskim2311** [comment]
Hi! @randimays I did not make the back button changes yet. I was going to do them with the review page update ticket. 

Form 293 changes were made for Air Force only, I believe I have made the changes to both I will pull main again just in case.

No ticket has been filed yet but definitely looks like a defect, saw it late yesterday was going to investigate it a bit more today it is like this on prod as well.  

I also still plan on doing a clean up PR when I have time to rename consts and add some documentation on functions/how things work. 

For the last flow, I will add a screenshot for that one I realized I did not provide a flow for that. Nice catch!

**Comment by chriskim2311** [comment]
@randimays I think this is ready for your to re-review. Also filed a DST ticket [here](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3119)

**Comment by randimays** [comment]
@chriskim2311 Thanks for the updates! I still see the `[branch of service]` placeholder - I wonder if maybe the review instance isn't updating?

Screenshot 2024-08-01 at 1 46 31 PM

**Comment by chriskim2311** [comment]
@randimays This is the same description from prod. Sorry for not answering that earlier. It confused me at first too but this is direct copy. I can make a note to ask CAIA about it.

**Comment by randimays** [comment]
@chriskim2311 Oohhhh ok. So maybe it was intentional (or is a very long-running prod bug 😬)

**Review by randimays — APPROVED** [review]

**Pull request merged**